builtin/logical/mysql/secret_creds.go

```diff
@@ -59,27 +67,58 @@ func (b *backend) secretCredsRevoke(
 		return nil, err
 	}
 
+	// Get the role name
+	rolenameRaw, ok := req.Secret.InternalData["rolename"]
+	if !ok {
+		return nil, fmt.Errorf("secret is missing rollname internal data")
```

Comment:
This isn't an error. Secrets issued before an upgrade will not have the role name. These should use the prior revocation logic.